src/pages/settings/settings.js

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  Card,
+  CardHeader,
+  CardFooter,
+  Button,
+  Page,
+  PageSection,
+  PageSectionVariants,
+  TextArea
+} from '@patternfly/react-core';
+import { withRouter } from 'react-router-dom';
+import { noop } from 'patternfly-react';
+import RoutedConnectedMasthead from '../../components/masthead/masthead';
+import { connect, reduxActions } from '../../redux';
+import Breadcrumb from '../../components/breadcrumb/breadcrumb';
+import { setUserWalkthroughs, getUserWalkthroughs } from '../../services/walkthroughServices';
+
+class SettingsPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { userWalkthroughs } = this.props;
+
+    this.state = {
+      value: userWalkthroughs || '',
+      isValid: true
+    };
+
+    getUserWalkthroughs().then(response => {
+      if (response.data) {
+        this.setState({
+          value: response.data,
+          isValid: true
+        });
+      } else {
+        this.setState({
+          value: '',
+          isValid: true
+        });
+      }
+    });
+  }
+
+  saveSettings = (e, value) => {
+    e.preventDefault();
+    const { history } = this.props;
+    setUserWalkthroughs(value);
+    history.push(`/`);
+  };
+
+  handleTextInputChange = value => {
+    this.setState(
+      {
+        value,
+        isValid: /^(?:https:\/\/)+([w.-]+)+github.com\/[\w\-._~:/?#[\]@!$&/'()*+,;=.]+$/.test(value)
+      },
+      () => {
+        if (this.state.value === '') {
+          this.setState({ isValid: true });
+        }
+
+        if (this.state.value.includes('\n')) {
+          const repoArray = this.state.value.split('\n');
+
+          for (let i = 0; i < repoArray.length; i++) {
+            if (/^(?:https:\/\/)+([w.-]+)+github.com\/[\w\-._~:/?#[\]@!$&/'()*+,;=.]+$/.test(repoArray[i])) {
+              this.setState({
+                isValid: true
+              });
+            } else if (repoArray[i] === '\n' || repoArray[i] === '') {
+              this.setState({
+                isValid: true
+              });
+            } else {
+              this.setState({
+                isValid: false
+              });
+            }
+          }
+        } else if (this.state.value === '') {
+          this.setState({ isValid: true });
+        } else {
+          this.setState({
+            isValid: /^(?:https:\/\/)+([w.-]+)+github.com\/[\w\-._~:/?#[\]@!$&/'()*+,;=.]+$/.test(value)
+          });
+        }
+      }
+    );
+  };
+
+  render() {
+    const { value, isValid } = this.state;
+
+    return (
+      <React.Fragment>
```

Comment:
I don't think you need React.Fragment here

Reply:
Yep, I removed it, works fine without.